code/modules/events/carp_migration.dm

```diff
@@ -30,10 +30,8 @@
 /datum/event/carp_migration/proc/spawn_fish(num_groups, group_size_min = 3, group_size_max = 5)
 	var/list/spawn_locations = list()
 
-	for(var/thing in GLOB.landmarks_list)
-		var/obj/effect/landmark/C = thing
-		if(C.name == "carpspawn")
-			spawn_locations.Add(C.loc)
+	for(var/thing in GLOB.carplist)
+		spawn_locations.Add(get_turf(thing))
```

Comment:
How exactly does this fix carp event?. The code seems to do the exact same thing?...

Reply:
Because carp landmarks are not kept in `GLOB.landmarks_list` but rather `GLOB.carplist`, so we want to check that not the full landmarks list.
This is true for a few other landmarks, if they have a specific GLOB list (as specified in the var `spawner_list` on initialize for said item) then they do not get added to the global one.
My thought here was to move away from searching one big-ass list for stuff, and instead have specific things on specific lists. When looking for carp landmarks we don't care about the syndie depot spawners for example.

Reply:
Ah, i see now...
The thing is that the carp spawner *does* get added to both `GLOB.carplist` AND `GLOB.landmarks_list`, but immediately after that it gets `qdel`'ed, and that deletes it from the `GLOB.landmarks_list` due to `/Destroy()` handling. But not from `GLOB.carplist`..  Because `GLOB.carplist` only holds the location, whereas `GLOB.landmarks_list` holds the landmark itself.

A bit convoluted, but it does indeed work. Thanks!